src/server/ocpp/services/OCPPService.ts

```diff
@@ -1135,31 +1135,36 @@ export default class OCPPService {
 
   private async checkNotificationEndOfCharge(tenant: Tenant, chargingStation: ChargingStation, transaction: Transaction) {
     // Transaction in progress?
-    if (transaction && !transaction.stop) {
-      // Has consumption?
-      if (transaction.numberOfMeterValues > 1 && transaction.currentTotalConsumptionWh > 0) {
-        // End of charge?
-        if (this.chargingStationConfig.notifEndOfChargeEnabled && transaction.currentTotalConsumptionWh > 0) {
-          // Battery full
-          if (transaction.currentStateOfCharge === 100) {
+    if (!transaction?.stop && transaction.currentTotalConsumptionWh > 0) {
+      // Check the battery
+      if (transaction.currentStateOfCharge > 0) {
+        // Check if battery is full (100%)
+        if (transaction.currentStateOfCharge === 100) {
+          // Send Notification
+          this.notifyEndOfCharge(tenant, chargingStation, transaction);
+        // Check if optimal charge has been reached (85%)
+        } else if (transaction.currentStateOfCharge >= this.chargingStationConfig.notifBeforeEndOfChargePercent) {
+          // Send Notification
+          this.notifyOptimalChargeReached(tenant, chargingStation, transaction);
+        }
+      // No battery information: check last consumptions
+      } else {
+        // Connector status must be 'Charging'
+        const connector = Utils.getConnectorFromID(chargingStation, transaction.connectorId);
+        if (connector.status === ChargePointStatus.CHARGING) {
```

Comment:
If the car is not consuming energy the connector status changes to SuspendedEV right? So I think this would need to be added here.

Reply:
What about checking SuspendedEV instead of Charging?

Reply:
In theory I would say this works. 
But sometimes the state of the connector also switches to SuspendedEVSE.
E.G.:
![image](https://user-images.githubusercontent.com/33114317/121886669-f72f1f80-cd15-11eb-8ef1-8f1eb383a619.png)
![image](https://user-images.githubusercontent.com/33114317/121886392-9dc6f080-cd15-11eb-8e06-ac7bc8ad505c.png)

I am not completely sure why it happens. Maybe the car is not able to charge at the min limit. Or it is a similar issue like with the Tesla Model 3, where it had something to do with mode 3 and the ocpp parameters.

Reply:
Ok, perfect, I just provided an enhancement in the code.
Let me know.

Reply:
I think we need status SuspendedEV here like you mentioned, because normally the charge is suspended by the EV and not by the EVSE, when the EV is fully charged.

I just wanted to mention, that we also have other cases, where the status says the EVSE suspends the charge. Sorry for the misunderstanding.

Reply:
No, you're right,  my bad! :-)
I just added the SuspendedEV status.
Let me know.

Reply:
Looks good to me know 👍